src/dnx/pal.win32.cpp

```diff
@@ -76,17 +81,41 @@ BOOL GetFullPath(LPCTSTR szPath, LPTSTR pszNormalizedPath)
     return TRUE;
 }
 
-HMODULE LoadNativeHost(LPCTSTR pszHostModuleName)
+int CallApplicationMain(const wchar_t* moduleName, const char* functionName, CALL_APPLICATION_MAIN_DATA* data, TraceWriter traceWriter)
 {
-    return LoadLibraryEx(pszHostModuleName, NULL, LOAD_LIBRARY_SEARCH_DEFAULT_DIRS);
-}
+    HMODULE hostModule;
+    try
+    {
+        hostModule = LoadLibraryEx(moduleName, NULL, LOAD_LIBRARY_SEARCH_DEFAULT_DIRS);
+        if (!hostModule)
+        {
+            throw std::runtime_error(std::string("Failed to load: ")
+                .append(dnx::utils::to_string(moduleName)));
+        }
 
-BOOL FreeNativeHost(HMODULE hHost)
-{
-    return FreeLibrary(hHost);
-}
+        traceWriter.Write(std::wstring(L"Loaded module: ").append(moduleName), true);
 
-FARPROC GetEntryPointFromHost(HMODULE hHost, LPCSTR lpProcName)
-{
-    return GetProcAddress(hHost, lpProcName);
+        auto pfnCallApplicationMain = (FnCallApplicationMain)GetProcAddress(hostModule, functionName);
```

Comment:
while it does not make much difference here "the thinking goes that casting is inherently an ugly operation and it requires ugly syntax" :smile: